src/ai/backend/test/templates/model_service/endpoint.py

```diff
@@ -0,0 +1,178 @@
+
+        params: dict[str, Any] = {
+            "image": image_dep.name,
+            "architecture": image_dep.architecture,
+            "model_id_or_name": model_service_dep.model_vfolder_name,
+            "initial_session_count": model_service_dep.replicas,
+            "resources": session_dep.resources,
+            "resource_opts": {},
+            "domain_name": domain_dep.name,
+            "group_name": group_dep.name,
+            "scaling_group": scaling_group_dep.name,
+            "model_mount_destination": model_service_dep.model_mount_destination,
+            "model_definition_path": model_service_dep.model_definition_path,
+            "cluster_mode": cluster_dep.cluster_mode,
+            "cluster_size": cluster_dep.cluster_size,
+            # TODO: Make `envs` required.
+            "envs": {
+                "TEST_KEY": "test_value",
+            },
+        }
+        params.update(self._extra_service_params())
+        return params
+
+    @abstractmethod
+    def _extra_service_params(self) -> dict[str, Any]:
+        raise NotImplementedError("Subclasses must implement the _extra_service_params method.")
+
+    # TODO:
+    async def _wait_until_all_inference_sessions_ready(
+        self,
+        client_session: AsyncSession,
+        endpoint_id: str,
+        replicas: int,
+        vfolder_id: UUID,
+    ) -> None:
+        """
+        Poll the service endpoint until every backing inference-session is RUNNING
+        and has the model vfolder mounted.
+        """
+        while True:
+            result = await client_session.Service(endpoint_id).info()
+            active_routes = result["active_routes"]
+            session_ids = [route["session_id"] for route in active_routes]
+
+            ready_session_cnt = 0
+            for session_id in session_ids:
+                session_info = await client_session.ComputeSession.from_session_id(
+                    UUID(session_id)
+                ).detail([
+                    session_node_fields["type"],
+                    session_node_fields["status"],
+                    session_node_fields["vfolder_mounts"],
+                ])
+
+                assert session_info["type"] == "inference", (
+                    f"Session type should be 'inference'. "
+                    f"Actual type: {session_info['type']}, session_id: {session_id}"
+                )
+                assert session_info["vfolder_mounts"] == [str(vfolder_id)], (
+                    f"Model vfolder should be mounted into the inference session. "
+                    f"Actual mounted vfolder: {session_info['vfolder_mounts']}, "
+                    f"session_id: {session_id}"
+                )
+                if session_info["status"] == "RUNNING":
+                    ready_session_cnt += 1
+
+            if ready_session_cnt >= replicas:
+                break
+
+            await asyncio.sleep(1)
```

Comment:
It seems like it's working this way because the current SSE model serving doesn't send events. Let's improve and change it later to deliver messages via SSE.